client/state/document-head/selectors.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { getSelectedSite, getGroupName } from 'state/ui/selectors';
+
+/**
+ * Returns the document title as set by the DocumentHead component or setTitle
+ * action.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Document title
+ */
+export function getTitle( state ) {
+	return state.documentHead.title;
+}
+
+/**
+ * Returns a count reflecting unread items.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Unread count (string because it can be e.g. '40+')
+ */
+export function getUnreadCount( state ) {
+	return state.documentHead.unreadCount;
+}
+
+/**
+ * Returns the formatted document title, based on the currently set title,
+ * unreadCount, and selected site.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?String}        Formatted title
+ */
+export function getFormattedTitle( state ) {
+	const siteSpecificGroups = [ 'sites', 'editor' ];
+	const title = getTitle( state );
+	const unreadCount = getUnreadCount( state );
+	const site = getSelectedSite( state );
+
+	let pageTitle = '';
+
+	if ( unreadCount ) {
+		pageTitle += '(' + unreadCount + ') ';
+	}
+
+	pageTitle += title;
+
+	// Display site name as title part only if we're in 'My Sites'
+	if ( includes( siteSpecificGroups, getGroupName( state ) ) && site ) {
+		pageTitle = appendSite( pageTitle, site );
+	}
+
+	if ( pageTitle ) {
+		pageTitle = decodeEntities( pageTitle ) + ' — WordPress.com';
```

Comment:
With all the conditions around concatenating parts, I wonder if we should just construct it as an array of pieces, then call `Array#join`.